src/JsonSchema/Data/StockMarketSchema.ts

```diff
@@ -0,0 +1,138 @@
+ * validation code uses "typeof" to check the data. "typeof functionName" is "function", not "object", so ajv sees it as
+ * invalid data. There are some ways to work around this problem, but all of them are complicated. In the end,
+ * JSONSchemaType is only a utility type. If our schema is designed and tested properly, after the validation, we can
+ * typecast the data.
+ */
+export const StockMarketSchema = {
+  $schema: "http://json-schema.org/draft-07/schema#",
+  type: "object",
+  patternProperties: {
+    "^(?!.*(lastUpdate|Orders|storedCycles|ticksUntilCycle))": {
+      type: "object",
+      additionalProperties: false,
+      properties: {
+        b: {
+          type: "boolean",
+        },
+        cap: {
+          type: "number",
+        },
+        lastPrice: {
+          type: "number",
+        },
+        maxShares: {
+          type: "number",
+        },
+        mv: {
+          type: "number",
+        },
+        name: {
+          type: "string",
+          enum: [...stockSymbolKeys],
+        },
+        otlkMag: {
+          type: "number",
+        },
+        otlkMagForecast: {
+          type: "number",
+        },
+        playerAvgPx: {
+          type: "number",
+        },
+        playerAvgShortPx: {
+          type: "number",
+        },
+        playerShares: {
+          type: "number",
+        },
+        playerShortShares: {
+          type: "number",
+        },
+        price: {
+          type: "number",
+        },
+        shareTxForMovement: {
+          type: "number",
+        },
+        shareTxUntilMovement: {
+          type: "number",
+        },
+        spreadPerc: {
+          type: "number",
+        },
+        symbol: {
+          type: "string",
+          enum: [...stockSymbolValues],
+        },
+        totalShares: {
+          type: "number",
+        },
+      },
+      required: [...stockKeys],
+    },
+  },
+  properties: {
+    lastUpdate: { type: "number" },
+    Orders: {
+      type: "object",
+      patternProperties: {
+        ".*": {
+          type: "array",
+          items: {
+            type: "object",
+            additionalProperties: false,
+            properties: {
+              pos: {
+                type: "string",
+                enum: [PositionType.Long, PositionType.Short],
+              },
+              price: {
+                type: "number",
+              },
+              shares: {
+                type: "number",
+              },
+              stockSymbol: {
+                type: "string",
+                enum: [...stockSymbolValues],
+              },
+              type: {
+                type: "string",
+                enum: [OrderType.LimitBuy, OrderType.LimitSell, OrderType.StopBuy, OrderType.StopSell],
+              },
+            },
+            required: [...orderKeys],
+          },
+        },
+      },
+      propertyNames: {
+        enum: [...stockSymbolValues],
```

Comment:
This also feels like it has potential to bite is on upgrade/downgrade cases. I would give so long hard thought to both loading old saves and loading newer save on an older game, vs what will happen if we load keys that are outside the factions enum.